src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -1970,40 +1971,61 @@ class RTDetrLoss(nn.Module):
-        weight_dict (`Dict`):
+        losses_weight_dict (`Dict`):
             Dictionary relating each loss with its weights. These losses are configured in RTDetrConf as
-            `weight_loss_vfl`, `weight_loss_bbox`, `weight_loss_giou`
+            `weight_loss_vfl`, `weight_loss_cross_entropy`, `labels_binary_cross_entropy`, `weight_loss_focal`,
+            `weight_loss_bbox`, `weight_loss_giou`.
         losses (`List[str]`):
             List of all the losses to be applied. See `get_loss` for a list of all available losses.
         alpha (`float`):
             Parameter alpha used to compute the focal loss.
         gamma (`float`):
             Parameter gamma used to compute the focal loss.
-        eos_coef (`float`):
-            Relative classification weight applied to the no-object category.
         num_classes (`int`):
             Number of object categories, omitting the special no-object category.
     """
 
-    def __init__(self, config):
+    def __init__(self, config: RTDetrConfig):
         super().__init__()
 
         self.matcher = RTDetrHungarianMatcher(config)
         self.num_classes = config.num_labels
-        self.weight_dict = {
-            "loss_vfl": config.weight_loss_vfl,
-            "loss_bbox": config.weight_loss_bbox,
-            "loss_giou": config.weight_loss_giou,
+        self.losses_weight_dict = {
+            "labels_varifocal": config.weight_loss_vfl,
+            "labels_cross_entropy": config.weight_loss_cross_entropy,
+            "labels_binary_cross_entropy": config.weight_loss_binary_cross_entropy,
+            "labels_focal": config.weight_loss_focal,
+            "bbox": config.weight_loss_bbox,
+            "giou": config.weight_loss_giou,
         }
-        self.losses = ["vfl", "boxes"]
-        self.eos_coef = config.eos_coefficient
-        empty_weight = torch.ones(config.num_labels + 1)
-        empty_weight[-1] = self.eos_coef
-        self.register_buffer("empty_weight", empty_weight)
-        self.alpha = config.focal_loss_alpha
-        self.gamma = config.focal_loss_gamma
-
-    def loss_labels_vfl(self, outputs, targets, indices, num_boxes, log=True):
+
+        loss_names = [
+            "labels_varifocal",
+            "labels_cross_entropy",
+            "labels_binary_cross_entropy",
+            "labels_focal",
+            "boxes",
+        ]
+        not_matched_losses = set(config.losses) - set(loss_names)
+        if not_matched_losses:
+            raise ValueError(
+                f"Unsupported losses: {not_matched_losses}, should be in {self.losses_weight_dict.keys()}"
+            )
+
+        self.losses = config.losses
+
+        # for multiclass classfication with cross-entropy loss
+        if "labels_cross_entropy" in self.losses:
+            class_weights = torch.ones(config.num_labels + 1)
+            class_weights[-1] = config.eos_coefficient
+            self.register_buffer("class_weights", class_weights)
```

Comment:
Does this name change causes backward compability?

Reply:
+ I think can we set both `labels_focal` and `labels_cross_entropy`? I don't think so

Reply:
yes,
1) as far as I see the loss is not a part of the model and is created for each iteration
2) at the moment `class_weights` is not used in loss computation, it only used in fixed losses

Reply:
> Can we set both labels_focal and labels_cross_entropy? I don't think so

Technically we can do it, but not sure how good the results would be 🙂